### ISSUE TITLE
Game written in 7.4.11 and updated to 8.0.3 won't launch on Steam Deck

### ISSUE THREAD

**Issue by mcordxeecee** [renpy/renpy#4270]
Hey there! I'm in the final stretch of development on a Ren'py project and I'm hitting the last hurdle: getting the game running on Steam Deck.

I've already read through both the excellent tutorial by Bob Conway and I have the game working on Windows desktop through Steamworks, but my game won't launch on the Deck. While set to launch with the .sh file included in the Windows/Mac/Linux build option, the Deck will simply display the logo for a bit before needing to be restarted. I've been working through the Steam Deck Guide and the aforementioned tutorial by Bob Conway, and through the course of this realized that all of the steam_deck specific functions in screens and options mentioned in the Ren'py on Deck guide are not present in my game's code, even though I am on 8.0.3. 

Here's the rub: In the course of (successfully!) troubleshooting an entirely separate issue with test builds crashing on certain machines, I updated Ren'py itself from 7.4.11 to 8.0.3. I know that this is strictly speaking **not** a recommended thing to do, but it was the only solution I could find that fixed the crashes. That fixed my issue then, but I'm concerned that it's gummed things up here. I next tried using the update_renpy_steam.py + steam sdk method to update an exported game build. Unfortunately even after making a new build, running the update, and updating through Steamworks, the same thing happens: if the game is launched through Steam on the Steam Deck, it hangs and the OS needs to be restarted. The Windows desktop versions still work fine.

To recap: The game was originally written in 7.4, then Ren'py was updated to 8.0.3 to solve a different issue, and now any build of the game with or without updating with the steam sdk refuses to launch on Deck. 

Is the best course of action here to simply start fresh with a new project in Ren'py 8.0.3 and port my work over? Or is there something else I can try?

Thanks for any help!

**Comment by renpytom** [comment]
Is there any debugging output produced? Can you give me access to the game files so I can try it on my Steam Deck?

**Comment by mcordxeecee** [comment]
> 

No debugging output sadly, it just hangs until the app is closed or the system is restarted. I'd be happy to send you the build though, what's the best way to get that to you?

**Comment by renpytom** [comment]
You can email the files to pytom@bishoujo.us. (If using google drive, renpytom@gmail.com.)

**Comment by mcordxeecee** [comment]
Thanks! I've sent the zip.

**Comment by mcordxeecee** [comment]
Whoops, didn't mean to hit close with comment, haha.

**Comment by mcordxeecee** [comment]
I did some more of my own testing today and made some lateral progress.

I remade the game in 8.0.3 (porting all my content over to a new project) and the game still refuses to launch on the Deck, and still performs the same loading-forever behavior until the system is restarted. 

To try and narrow down what the issue was, I tried making a brand new Ren'py project in 8.0.3 and creating a build of the Eileen tutorial startup just to try and see if there's something in my game's code that's causing the Deck issues. Curiously, the exact same thing happens.

**Comment by mcordxeecee** [comment]
After a bit more digging, I think I have discovered what the problem was. It was likely not an issue with the builds or moving between Ren'py versions--although I did make a new Linux-specific one which launched on Ubuntu and the Deck in developer mode just fine--but an issue with the Steam Deck's cache!

For anyone in future struggling with similar issues, here's what to do:

1. On the Steam Deck, navigate to the gear icon on the game's page in your library and press A.
2. Go to the "Compatibility" tab.
3. Check **and uncheck** the available box.
4. Uninstall the game.
5. Reinstall the game. It should now download a build and use the correct launch path configuration.

The game boots and runs normally now. When I initially downloaded the build before switching to the beta branch in Steam to test it, it must have downloaded the Windows configuration and launch settings, and even simple re-installs wouldn't get it to recognize the proper branch to be installing from.
